petab/visualize/helper_functions.py

```diff
@@ -373,45 +371,75 @@ def handle_dataset_plot(i_visu_spec,
         conditions = exp_conditions[ind_cond][indep_var]
 
     # retrieve measurements from dataframes
-    measurement_to_plot = get_data_to_plot(vis_spec, exp_data, sim_data,
-                                           uni_condition_id, i_visu_spec,
-                                           col_name_unique)
+    measurement_to_plot = get_data_to_plot(plot_spec, exp_data, sim_data,
+                                           uni_condition_id, col_name_unique)
 
     # check, whether simulation should be plotted
-    plot_sim = True
-    if sim_data is None:
-        plot_sim = False
+    plot_sim = sim_data is not None
 
     # plot data
     nan_set = all([np.isnan(val) for val in measurement_to_plot['mean']])
     if not nan_set:
-        ax = plot_lowlevel(vis_spec, ax, i_row, i_col, conditions,
-                           measurement_to_plot, ind_plot, i_visu_spec,
-                           plot_sim)
+        plot_lowlevel(plot_spec, ax, conditions, measurement_to_plot, plot_sim)
 
     # Beautify plots
-    ax[i_row, i_col].set_xlabel(
-        vis_spec.xLabel[i_visu_spec])
-    ax[i_row, i_col].set_ylabel(
-        vis_spec.yLabel[i_visu_spec])
+    ax.set_xlabel(
+        plot_spec.xLabel)
+    ax.set_ylabel(
+        plot_spec.yLabel)
+
+
+def matches_plot_spec(df: pd.DataFrame,
+                      col_id: str,
+                      x_value: Union[float, str],
+                      plot_spec: pd.Series) -> pd.Series:
+    """
+    constructs an index for subsetting of the dataframe according to what is
+    specified in plot_spec.
+
+    Parameters:
+        df:
+            pandas data frame to subset, can be from measurement file or
+            simulation file
+        col_id:
+            name of the column that will be used for indexing in x variable
+        x_value:
+            subsetted x value
+        plot_spec:
+            visualization spec from the visualization file
 
-    return ax
+    Returns:
+        index:
+            boolean series that can be used for subsetting of the passed
+            dataframe
+    """
+    subset = (
+        (df[col_id] == x_value) &
+        (df[DATASET_ID] == plot_spec[DATASET_ID])
+    )
+    if plot_spec[Y_VALUES] == '':
+        if len(df.loc[subset, OBSERVABLE_ID].unique()) > 1:
+            ValueError(
+                f'{Y_VALUES} must be specified in visualization table if '
+                f'multiple different observables are available.'
+            )
+    else:
+        subset &= (df[OBSERVABLE_ID] == plot_spec[Y_VALUES])
+    return subset
```

Comment:
before one could use the `OBSERVABLE_ID` column also as index when handing over `pandas.Dataframes` instead of tsv-tables to `plot_data_and_simulation()`. This will not work anymore.
I actually think this is an improvement, since it clarifies input-format, but was just wondering if this should be noted in  `plot_data_and_simulation()`'s docstrings or the next release note

Reply:
Not sure what you mean. In my hands `OBSERVABLE_ID` didn't have any noticeable effect at all.

Reply:
yes, that is why it was also working with a dataframe which had the `OBERSAVBLE_ID` column not as regular column but as index.
This is not a feature which should be supported, but I was just thinking if this change should be indicated somewhere.